pkg/generate/app/cmd/newapp.go

```diff
@@ -901,7 +901,7 @@ func optionallyValidateExposedPorts(config *AppConfig, repositories app.SourceRe
 		return nil
 	}
 
-	if config.Strategy != "docker" {
+	if len(config.Strategy) > 0 && config.Strategy != "docker" {
```

Comment:
Yes, but in this case the type will be `docker` (and not `dockerfile`). In other words, empty strategy is the same as `docker` strategy.